core/state/statedb.go

```diff
@@ -372,12 +372,16 @@ func (self *StateDB) deleteStateObject(stateObject *StateObject) {
 // Retrieve a state object given my the address. Returns nil if not found.
 func (self *StateDB) GetStateObject(addr common.Address) (stateObject *StateObject) {
 	// Prefer 'live' objects.
+	self.lock.Lock()
```

Comment:
Just a suggestion: if perhaps lock was sync.RWMutex instead of sync.Mutex, then we could do a Read Lock and Read Unlock which are cheaper and less contentious than a plain Lock and Unlock so
```go
self.lock.RLock()
...
self.lock.RUnlock()
```
when doing reads and then normal lock and unlock when doing write.